### PR TITLE
feat: additional parameters for anf

### PR DESCRIPTION
**Pull request by bobmclane999**
## Description
Additional parameters for - 

- snapshot policies
- backup policies
- backup vaults and backups
- encryptionKeySource - Microsot.KeyVault or Microsoft.NetApp
- Latest API's 2024-03-01 as per NetApp releases
- fix ternary statements

## Pipeline Reference

| Pipeline |
| -------- |
[avm.res.net-app.net-app-account](https://github.com/bobmclane999/bicep-registry-modules/actions/workflows/avm.res.net-app.net-app-account.yml)
|          |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by bobmclane999** [comment]
Hi @jtracey93 / @fbinotto , please can you review and hopefully approve my PR?
Thanks
Shaun

**Review by fbinotto — CHANGES_REQUESTED** [review]
Please provide green deployment badge to confirm this was successfully deployed.

File: .vscode/extensions.json
Comment (AlexanderSehr) [inline .vscode/extensions.json:4]: Intended?

File: avm/res/net-app/net-app-account/capacity-pool/volume/main.bicep
Comment (AlexanderSehr) [inline avm/res/net-app/net-app-account/capacity-pool/volume/main.bicep:262]: Interesting. This one (and the one below) 'should' be their own child-module of 'netAppAccounts', that would ultimately be published as its own module. This would require you to move all parameters specific to this resource into that nested child-module

Comment (bobmclane999) [reply avm/res/net-app/net-app-account/capacity-pool/volume/main.bicep:266]: Hey @AlexanderSehr  understand however under some time constraints before my vacation tomorrow is it OK to gain approval on this PR now and I can come back to this and do once I return? Thanks Shaun

Comment (bobmclane999) [reply avm/res/net-app/net-app-account/capacity-pool/volume/main.bicep:266]: @fbinotto  - Hi green badge now attached.  As above if this PR can be approved as is and once back from vacation I'll look to put the backup policies and snapshot policies into their own file...  unless someone else has some time to do while I'm OOTO ;)

**Comment by bobmclane999** [comment]
> Please provide green deployment badge to confirm this was successfully deployed.

Green badge attached.  Will look into Alexander's request to split out the backup and snapshot policies on my return from holiday.

**Comment by bobmclane999** [comment]
Hi @fbinotto @AlexanderSehr @jtracey93  - green badge now attached. 
As above if this PR can be approved as is and once back from vacation I'll look to put the backup policies and snapshot policies into their own files... unless someone else has some time to do while I'm OOTO ;P
Thanks for all your help!
Cheers
Shaun

**Comment by fbinotto** [comment]
Merging, @bobmclane999 please split backup policies and snapshot policies into their own files once you are back.

**Review by fbinotto — DISMISSED** [review]
LGTM

**Comment by bobmclane999** [comment]
> Approved, @bobmclane999 please split backup policies and snapshot policies into their own files once you are back. @AlexanderSehr @jtracey93 please approve and merge.

Thanks for the flexibility guys, much appreciated!  Will add a backlog Task in our board to sort once back... tyvm
@jtracey93 / @AlexanderSehr  - If you could please approve also that would be awesome then I can get my ANF volumes deployed in Prod :)...
Regards
Shaun

File: .vscode/extensions.json
Comment (AlexanderSehr) [reply .vscode/extensions.json:4]: Not sure why this was resolved. This change is the one currently blocking the merge 😄 
@fbinotto's approval is enough for the module to be merged, but if it's anything outside a module (like in this case a settings file), this is when a maintainer must approve. Please just undo this change and we can get the PR in @bobmclane999  😉

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**
